### PR TITLE
Functional !queue for tunes. Cosmetic changes.

### PR DESCRIPTION
**Pull request by matthew-morrison**
Change commands to use embeddings.
Add documentation to more commands.
New queue command for tunes.
Congregate memes.
Improvements probably.

### CODE REVIEW

File: csss-minion.py [hunk 1]
```diff
@@ -225,6 +231,14 @@ def updateLevel(change, experience, currLevel):
         return True
     return False  
 
+async def embed_this_for_me(text, ctx):
```
Comment (henrymzhao) [inline csss-minion.py]: Both y'all @brenfan and @matthew-morrison  wrote the exact same function. I'm not gonna have redundant code up in here, so you guys hash it out and fix it before I merge.

File: csss-minion.py [hunk 2]
```diff
@@ -244,11 +258,12 @@ def updateLevel(change, experience, currLevel):
...
             print('Failed to load extension {}\n{}'.format(extension, exc))
 
-@bot.command()
-async def cogs():
+@bot.command(pass_context=True)
+async def cogs(ctx):
     """Lists the currently loaded cogs."""
     cogs = list(bot.cogs.keys())
-    await bot.say("\n".join(cogs)) 
+    await bot.embed_this_for_me("\n".join(cogs), ctx)
```
Comment (henrymzhao) [inline csss-minion.py]: gud

File: memes.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+class Memes:
+
+    def __init__(self, bot):
+        self.bot = bot
+        
+# also this http://i.imgur.com/dmqYSvu.jpg
+    @commands.command(pass_context=True)
+    async def impeach(self,ctx):
+        """Impeach our dear leader"""
```
Comment (henrymzhao) [inline memes.py]: no

File: csss-minion.py (hunk 1 above)
Comment (matthew-morrison) [inline csss-minion.py]: Referring to the announce and em commands? I see nothing which allows outside cogs to use these functions.

**Pull request merged**